src/rgw/rgw_s3gw_telemetry.cc

```diff
@@ -199,23 +193,50 @@ void S3GWTelemetry::Version::decode_json(JSONObj* obj) {
   JSONDecoder::decode_json("ReleaseDate", release_date, obj, true);
 }
 
-std::optional<std::vector<S3GWTelemetry::Version>>
-S3GWTelemetry::parse_upgrade_response(bufferlist& response) const {
+bool S3GWTelemetry::parse_upgrade_response(
+    bufferlist& response, std::chrono::milliseconds* out_update_interval,
+    std::vector<Version>* out_versions
```

Comment:
Why using pointers for those 2 return values? 
Code is not checking if passed value is `nullptr` before using them, so I guess we are always going to set their values.

I suggest using references for them.